feat: Creates the Recipe App application

gvgesanto2 opened the pull request:
## Description

**Task:** https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/buisness_requirements.md

This PR makes the following changes:
-  Creates the ``Recipe App`` application using Ruby on Rails.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [X] Our code follows the style guidelines of this project (no linters errors)
- [X] We have checked our code and corrected any misspellings
- [X] We have deleted any unused code as comments in our codebase

gvgesanto2 committed:
feat: Set up the initial Rails application and linters

gvgesanto2 committed:
feat: Build the database schema for the project

gvgesanto2 committed:
feat: Create the app models

gvgesanto2 committed:
feat: Add the Devise library for authentication

gvgesanto2 committed:
feat: Set up the app routes

gvgesanto2 committed:
feat: Set up Sass and the initialCSS architecture

gvgesanto2 committed:
feat: Add CanCanCan library for authorization

gvgesanto2 committed:
feat: Add the Header initial structure

gvgesanto2 committed:
chore: Set up RSpec and Capybara testing libraries

gvgesanto2 committed:
fix: Fix linters errors

gvgesanto2 committed:
feat: Add description field to Inventories table

gvgesanto2 committed:
feat: Add the project's color pallete

gvgesanto2 committed:
fix: Fix linters errors

gvgesanto2 committed:
Merge pull request #25 from gvgesanto2/setup

feat: Set up the Rails initial project

desorgui committed:
generate scaffold for inventory

gvgesanto2 committed:
feat: Create and style the recipes index page

gvgesanto2 committed:
feat: Add the photo column to the Recipes table

desorgui committed:
Add inventory_foods controller and view

desorgui committed:
Add N+1 queries

gvgesanto2 committed:
feat: Add the c-card and c-section-title CSS classes

desorgui committed:
Add description to inventories table

desorgui committed:
Merge branch 'dev' into inventories

desorgui committed:
Fix linters

desorgui committed:
Fix stylelint

desorgui committed:
Add styles to inventories index page

gvgesanto2 committed:
feat: Add the functionality to upload recipe images

desorgui committed:
Add styles to inventories

desorgui committed:
Implement delete inventory food button

desorgui committed:
Add styles to form input

desorgui committed:
resolve conflict

gvgesanto2 committed:
feat: Style the new recipe form

desorgui committed:
Add styles to form

gvgesanto2 committed:
feat: Create the show recipe page

desorgui committed:
Add controller  for shopping list

desorgui committed:
Use recipe id instead of food idin shopping list controller

gvgesanto2 committed:
feat: Create the add and deleteingredients functionality

gvgesanto2 committed:
feat: Create the generate shopping list popup

desorgui committed:
Fix rubocop errors

desorgui committed:
Fix stylelint errors

gvgesanto2 committed:
feat: Manage access to the recipes resource

gvgesanto2 committed:
feat: Add header dropdowns

gvgesanto2 committed:
fix: Fix linters errors

gvgesanto2 committed:
Merge pull request #28 from gvgesanto2/recipes

feat: Create the Recipe pages

gvgesanto2 committed:
Merge branch 'dev' into inventories

desorgui committed:
Merge pull request #26 from gvgesanto2/inventories

Inventories

gvgesanto2 committed:
feat: Add shopping list route

desorgui committed:
Implement shopping list

gvgesanto2 committed:
feat: Create the javascript Shoplist controller

gvgesanto2 committed:
Merge remote-tracking branch 'origin/shopping-list' into navbar

gvgesanto2 committed:
fix: Fix bug in shoplist_controller.js file

desorgui committed:
Add total price in shopping list

gvgesanto2 committed:
feat: Finish styling the navbar

gvgesanto2 committed:
fix: Allow logged-in users to generate shopping list from public recipes

gvgesanto2 committed:
Merge remote-tracking branch 'origin/shopping' into navbar

gvgesanto2 committed:
feat: Change some styles in the shopping list page

gvgesanto2 committed:
fix: Fix linters errors

gvgesanto2 committed:
Merge pull request #29 from gvgesanto2/navbar

feat: Create the application's Navbar

gvgesanto2 committed:
fix: Fixing merge conflicts

gvgesanto2 committed:
fix: Fix bugs in the foods controller and model

gvgesanto2 committed:
fix: Fix linters errors

gvgesanto2 committed:
docs: Update README.md file

gvgesanto2 committed:
Merge pull request #30 from gvgesanto2/merge-conflicts

fix: Fix conflicts and integrate the Food-list-page branch

Bushido-brown committed:
Update files

elisha2kyakpo1 commented on app/views/devise/mailer/confirmation_instructions.html.erb:
```diff
@@ -0,0 +1,5 @@
+<p>Welcome <%= @email %>!</p>
+
+<p>You can confirm your account email through the link below:</p>
+
+<p><%= link_to 'Confirm my account', confirmation_url(@resource, confirmation_token: @token) %></p>
```
- [ ] I failed to register because the app was asking me to confirm my email address, and unfortunately, I didn't receive any email for me to confirm, note: I used my actual (real) email. This prevented me from testing the app.

![email](https://user-images.githubusercontent.com/53124548/196015242-689c4897-0a0f-4093-864b-434da1fd7542.png)

gvgesanto2 replied:
Hi @elisha2kyakpo1,

As this application is not in production yet, it's not set up to actually send emails. So, to register a new account and be able to test the application, you can check the terminal where you are running the server. The message the user is supposed to receive by email will be printed there, so you just need to click on the confirmation link:

![terminal](https://user-images.githubusercontent.com/64566209/196016197-83a533c8-8c5d-49b7-9904-7174d2d241d3.png)

gvgesanto2 replied:
I also configured the application to send a fake smtp email to mailcatcher. So, alternatively, you could install the [MailCatcher](https://mailcatcher.me/) program with the command:
```
$ gem install mailcatcher
```
Doing this you will be able to catch the emails sent by this application on a more friendly interface:
![mailcatcher](https://user-images.githubusercontent.com/64566209/196016406-20e7df78-1dfb-4951-82de-aca653371ba9.png)

elisha2kyakpo1 reviewed (changes requested):
Hi 👋 team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- The PR title and description looks professional✔️
- Descriptive commit messages✔️
- You have a descriptive readme file✔️
- No linter errors✔️

## Required Changes ♻️
_Check the comments under the review._

- [ ] Kindly add unit and integration tests, because it's a requirement for this project.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

gvgesanto2 committed:
test: Add tests

gvgesanto2 committed:
test: Add tests for the Inventory, User, and Food models

gvgesanto2 committed:
fix: Fix linters errors

gvgesanto2 committed:
Merge pull request #32 from gvgesanto2/tests

test: Add tests

gvgesanto2 commented:
## Description

**Task:** https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/buisness_requirements.md

This PR addresses the change requested by the [last review](https://github.com/gvgesanto2/microverse-recipe-app/pull/31#pullrequestreview-1143245239).

## Type of change

- [ ] Performance (non-breaking change which fixes performance issues)
- [X] Test (change related to testing)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [X] My code follows the style guidelines of this project (no linters errors)
- [X] I have checked my code and corrected any misspellings
- [X] I have deleted any unused code as comments in my codebase

arslanbisharat reviewed (approved):
Hi @gvgesanto2,
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

arslanbisharat reviewed (approved):
Hi @gvgesanto2 & @elisha2kyakpo1,
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
